### PR TITLE
7. POST /api/reviews/:review_id/comments

### PR DESCRIPTION
**Pull request by MPlace1**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -168,4 +168,71 @@ describe('/api/reviews/:review_id/comments', () => {
...
+
+            return request(app)
+                .post("/api/reviews/1/comments")
+                .send(newComment)
+                .expect(400)
+                .then(({ body }) => {
+                    expect(body.msg).toEqual("Invalid comment");
+                });
+        });
+    });
```
Comment (Rayhaan2k9) [inline __tests__/app.test.js:263]: Another test or 2 will be useful here! What if a valid comment is made to an article that doesn't exist? Also, anybody post comments or should only the registered users be able to comment?

File: models/bg.model.js [hunk 2]
```diff
@@ -54,4 +54,35 @@ exports.fetchReviewComments = (review_id) => {
...
+};
+
+exports.addReviewComment = (review_id, reqBody) => {
+    const { username, body } = reqBody;
+    if (Object.keys(reqBody).length === 2) {
+        const reqKeys = [
+            "username",
+            "body",
+        ];
+        console.log(reqBody)
```
Comment (Rayhaan2k9) [inline models/bg.model.js]: Please remove the console log

File: models/bg.model.js [hunk 3]
```diff
@@ -54,4 +54,35 @@ exports.fetchReviewComments = (review_id) => {
...
+        bool = Object.keys(reqBody).every((key) => reqKeys.includes(key));
+        console.log(bool)
```
Comment (Rayhaan2k9) [inline models/bg.model.js]: And this one

File: models/bg.model.js [hunk 4]
```diff
@@ -54,4 +54,35 @@ exports.fetchReviewComments = (review_id) => {
...
+        if (!bool) {
+            return Promise.reject({ status: 400, msg: "Invalid comment" });
+        }
+        console.log(2)
```
Comment (Rayhaan2k9) [inline models/bg.model.js]: Also this one

File: models/bg.model.js [hunk 5]
```diff
@@ -54,4 +54,35 @@ exports.fetchReviewComments = (review_id) => {
...
+        return db
+            .query(
+            `INSERT INTO comments
+            (author, body, review_id)
+            VALUES ($1, $2, $3)
+            RETURNING *;`,
+                [username, body, review_id]
+            )
+            .then((comment) => {
+                console.log(comment.rows)
```
Comment (Rayhaan2k9) [inline models/bg.model.js]: And this one too

**Review by Rayhaan2k9 — CHANGES_REQUESTED** [review]
This looks great but is just missing 1 or 2 tests. Please add these as well as the functionality and then resubmit the PR 🙂

**Review by Slyose — APPROVED** [review]
Great feedback response, feel free to merge.

**Pull request merged**